src/components/sortable_table/row.js

```diff
@@ -0,0 +1,249 @@
+var _ = require('lodash');
+var Estimator = require('../estimator');
+var Status = require('../status');
+var Tags = require('../tags');
+var TagEditor = require('../tag_editor');
+var SortableStyles = require('../../styles/sortable_table');
+var moment = require('moment');
+
+/*
+ * Renders a single table row for displaying item data.
+ * Row accepts expanded prop for controlling whether the row appears
+ * condensed (default) or optionally expanded (if using row with the sortable TableHeader.
+ * Toggling between the two states happens via the Expander element in TableHeader.)
+ */
+
+var abbreviateUsername = function (user) {
+  var name = user.first_name;
+  if (user.last_name) {
+    name += " " + user.last_name[0] + ".";
+  }
+  return name;
+};
+
+var TableRow = React.createClass({
+  propTypes: {
+    model: React.PropTypes.object.isRequired,
+    columns: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
+    expanded: React.PropTypes.string,
+    modelChangerUtilities: React.PropTypes.object,
+    navigatorUtility: React.PropTypes.object,
+    isBulkEditable: React.PropTypes.bool,
+    onBulkSelect: React.PropTypes.func
+  },
+
+  mixins: [SortableStyles],
+
+  getDefaultProps: function() {
+    return {
+      expanded: 'condensed',
+      modelChangerUtilities: {},
+      onBulkSelect: _.noop()
+    };
+  },
+
+  getInitialState: function() {
+    return {
+      hover: false
+    };
+  },
+
+  bulkSelectClicked: function(ev) {
+    // WIP(fw) This will likely change.
+    ev.stopPropagation();
+    this.props.onBulkSelect(this.props.model);
+  },
+
+  render: function() {
+    /*
+     * This is so gross. Can we find a way around this column checking?
+     * Do all columns need to be optional?
+     */
+    var modelId = [this.props.model.product.id, this.props.model.number];
+    var condensed = this.props.expanded === 'condensed' ? true : false;
+    var itemType = this.props.model.type;
+
+    var rowClass = "table-row " + this.props.expanded;
+    var rowStyles = _.extend({}, SortableStyles.row[itemType],
+      !condensed ? SortableStyles.row.expanded : null);
+
+    // "matched": item is grouped with parent/subitems.
+    // "nonMatching": item doesn't fit collection filters, but is included to match w/subitems that do.
+    if (this.props.model.isMatched) {
+      rowStyles = _.extend({}, rowStyles, SortableStyles.row.matched,
+        this.props.model.isNonMatching ? SortableStyles.row.nonMatching : null);
+    }
+
+    var wrapperStyles = condensed ? SortableStyles.cell.condensed : SortableStyles.cell.expanded;
+
+    // Lookup hash for easy column-type inclusion checking.
+    // TODO(fw): find a better way to do this.
+    var columns = _.object(this.props.columns, this.props.columns);
+    var possibleCells = {
+      control: this.props.isBulkEditable ? this.buildControlCell(wrapperStyles) : null,
+      product: columns.product ? this.buildProductCell(wrapperStyles) : null,
```

Comment:
It seems like this might be better if we iterate over a list of known properties of `columns` and relying on naming convention for the `buildX` components, rather than checking each one with a ternary.